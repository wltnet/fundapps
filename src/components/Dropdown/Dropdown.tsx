import * as React from "react";
import { Props } from ".";

const Dropdown = ({ list, onChange }: Props): React.ReactElement => (
  <select name="source" id="source" onChange={onChange}>
    <option value="Filter by source">Filter by source</option>
    {list.sort().map((source) => (
      <option key={source} value={source}>
        {source}
      </option>
    ))}
  </select>
);

export default Dropdown;
