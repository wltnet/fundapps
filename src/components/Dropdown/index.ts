import Dropdown from "./Dropdown";
import * as React from "react";

export interface Props {
  list: string[];
  onChange: (event: React.ChangeEvent) => void;
}

export default Dropdown;
