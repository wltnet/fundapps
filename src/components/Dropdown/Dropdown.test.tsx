import React from "react";
import { shallow } from "enzyme";
import Dropdown from "./Dropdown";

describe("<Dropdown />", () => {
  it("should match the snapshot", () => {
    const props = {
      list: ["CNN", "BBC"],
      onChange: jest.fn(),
    };
    const wrapper = shallow(<Dropdown {...props} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("option").at(1).text()).toBe("BBC");
    expect(wrapper.find("option").at(2).text()).toBe("CNN");
  });
});
