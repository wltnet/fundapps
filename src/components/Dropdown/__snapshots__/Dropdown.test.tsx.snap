// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> should match the snapshot 1`] = `
<select
  id="source"
  name="source"
  onChange={[MockFunction]}
>
  <option
    value="Filter by source"
  >
    Filter by source
  </option>
  <option
    key="BBC"
    value="BBC"
  >
    BBC
  </option>
  <option
    key="CNN"
    value="CNN"
  >
    CNN
  </option>
</select>
`;
